clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -58,31 +56,82 @@ static RegisterType getRegisterType(ResourceClass RC) {
   llvm_unreachable("unexpected ResourceClass value");
 }
 
-static RegisterType getRegisterType(StringRef Slot) {
+// Converts the first letter of string Slot to RegisterType.
+// Returns false if the letter does not correspond to a valid register type.
+static bool convertToRegisterType(StringRef Slot, RegisterType *RT) {
```

Comment:
I've considered it, but since we `convertToRegisterType` does not create an error string or code that we should report I just kept it simple.

Reply:
Well, I tried to do it with `Expected<RegisterType>` and while does make the `convertToRegisterType` function smaller, checking of the results got more complicated because LLVM forces the error string/whatever to be checked. I'll leave it as is.